pkg/client/cache/store.go

```diff
@@ -150,10 +232,22 @@ func (c *cache) Replace(list []interface{}) error {
 	c.lock.Lock()
 	defer c.lock.Unlock()
 	c.items = items
+
```

Comment:
Can you build the index before the lock and then swap it in to limit contention?  (Similar to above)